libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/CoreBot.cs

```diff
@@ -0,0 +1,99 @@
+        /// <remarks>
+        /// This method calls other methods in this class based on the type of the activity to
+        /// process, which allows a derived class to provide type-specific logic in a controlled way.
+        ///
+        /// In a derived class, override this method to add logic that applies to all activity types.
+        /// Add logic to apply before the type-specific logic before the call to the base class
+        /// <see cref="OnTurnAsync(ITurnContext, CancellationToken)"/> method.
+        /// Add logic to apply after the type-specific logic after the call to the base class
+        /// <see cref="OnTurnAsync(ITurnContext, CancellationToken)"/> method.
+        /// </remarks>
+        /// <seealso cref="ActivityHandler.OnMessageActivityAsync(ITurnContext{IMessageActivity}, CancellationToken)"/>
+        /// <seealso cref="ActivityHandler.OnConversationUpdateActivityAsync(ITurnContext{IConversationUpdateActivity}, CancellationToken)"/>
+        /// <seealso cref="ActivityHandler.OnMessageReactionActivityAsync(ITurnContext{IMessageReactionActivity}, CancellationToken)"/>
+        /// <seealso cref="ActivityHandler.OnEventActivityAsync(ITurnContext{IEventActivity}, CancellationToken)"/>
+        /// <seealso cref="ActivityHandler.OnUnrecognizedActivityTypeAsync(ITurnContext, CancellationToken)"/>
+        /// <seealso cref="Activity.Type"/>
+        /// <seealso cref="ActivityTypes"/>
+        public virtual async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await _dialogManager.OnTurnAsync(turnContext, cancellationToken: cancellationToken).ConfigureAwait(false);
+            await _conversationState.SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+            await _userState.SaveChangesAsync(turnContext, false, cancellationToken).ConfigureAwait(false);
+        }
+
+        private DialogManager CreateDialogManager(IServiceProvider services, IOptions<CoreBotOptions> options)
+        {
+            var resourceExplorer = services.GetRequiredService<ResourceExplorer>();
+            var telemetryClient = services.GetService<IBotTelemetryClient>();
+
+            Resource rootDialogResource = resourceExplorer.GetResource(options.Value.RootDialog); 
+            var rootDialog = resourceExplorer.LoadType<AdaptiveDialog>(rootDialogResource);
+
+            var dialogManager = new DialogManager(rootDialog)
+                .UseResourceExplorer(resourceExplorer)
+                .UseLanguageGeneration()
+                .UseLanguagePolicy(new LanguagePolicy(options.Value.DefaultLocale ?? DefaultLocale));
+
+            if (telemetryClient != null)
+            {
+                dialogManager.UseTelemetry(telemetryClient);
+            }
+
+            dialogManager.InitialTurnState.Set(services.GetRequiredService<BotFrameworkClient>());
```

Comment:
What would be the process for a developer who wants to add something to InitialTurnState?

Or, if someone wants to use ContinueConversation dialog and   QueueStorage?

```cs
var queueStorage = dc.Context.TurnState.Get<QueueStorage>() ?? throw new NullReferenceException("Unable to locate QueueStorage in HostContext");
var receipt = await queueStorage.QueueActivityAsync(continuationActivity, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Reply:
Few things here. Some context first. On one hand, there have been discussions about what is the relation between bot, activity handler, etc. Before setting and communicating a pattern for those scenarios maybe we need to resolve the patterns in relationships between these classes. 

Other piece, is that the current PR and R12 preview does not have the goal to make a generic runtime that can do any arbitratily complex scenario, but a runtime for composer that supports all current composer features plus some specific ones that were planned (component model, configurable adapters, etc). Eventually during R13 coding and onwards, this can shift towards being a more general runtime but that has not been a business goal of this code as of yet. 

With that being said, in R13 we should come up with a concrete list of scenarios like the 2 you mentioned, and more, and make a design of how to support them.

As of the immediate solution, i.e., composer users who opt in a preview runtime to do some previewing of these features, they can create and inject their own bot that behaves however they want I guess. Again, this is not a problem that was a goal to solve right now but happy to hear and / or brainstorm ideas on how it should be solved and file an issue for R13.